### ISSUE TITLE
Mismatch compressed/decompressed NSP

### ISSUE THREAD

**Issue by proconsule** [nicoboss/nsz#101]
Hi nicoboss, i am working to fully support nsz in NSP Indexer.
I have right now a full decompression routine that devliver the same result as nsz command line.
But i noted that the original NSP and the decompressed NSP have a diff.
both works but i think there is a mismatch in offsets into the decompressed one

Original NSP
 pfs0:/0100ea80032ea0000000000000000004.cert 000000000000-000000000700

Compressed/Decompressed NSP
pfs0:/0100ea80032ea0000000000000000004.cert 000000007df0-0000000084f0

Edit:

I modified the way i recreate the main PFS0 and now i have perfect match from Original to Compressed/Decompressed
same sha256sum!
i just started to put files from fileBodyOffset instead of starting from fileBodyOffset+ dataoffset (that in nsz is shifted)

**Comment by josearr** [comment]
This might be related to what I just commented here: https://gbatemp.net/threads/nsz-homebrew-compatible-nsp-xci-compressor-decompressor.550556/post-9622769

The hashes of .nsp files before compressing and after decompressing them don't match.

**Comment by LittleWhite-tb** [comment]
By curiousity, it seems that the problem is only when decompressing the NSZ with nsz. When adding a file to the `Pfs0Stream`, it will use `self.f.tell()` as the offset. But the PFS0 file is already at a certain position (0x800), because the constructor is doing `self.f.seek(self.headerSize)`.
It can be naively fixed in `getHeader()` by using `self.headerSize` as `headerSize`, instead of the `0x10 + len(self.files) * 0x18 + len(stringTable)`

The good new is that the NSZ is correct, just the recreation of the NSP from NSZ is a bit wrong.

**Comment by nicoboss** [comment]
This is fixed in the latest [NSZ v4.3.0](https://github.com/nicoboss/nsz/releases) release. The padding of the source NSP is now kept during compression/decompression making them always bit-identical (especially if --keep-delta is specified). The option --remove-padding was added to remove the padding of existing NSP/NSZ files to make them nxdumptool/no-intro compliant. This was done by applying the exact fix @LittleWhite-tb described.
```python
container.getHeaderSize() if removePadding else container.getFirstFileOffset()
```
@proconsule and @LittleWhite-tb Thanks a lot for all your investigations regarding this issue.
